1.16.5 Fabric

CoolMineman opened the pull request:
Fixes #32 
Fixes #31 
Fixes #26 (Also fixes Slow Falling)
Related #21

CoolMineman committed:
fix compiling

CoolMineman committed:
port to 1.16.1

CoolMineman committed:
fix minecraft_ApplyGravity

CoolMineman committed:
fix sharking

CoolMineman committed:
fix sharking water issues

CoolMineman committed:
clean up mixins

CoolMineman committed:
remove useless code

CoolMineman committed:
fix surface tension

CoolMineman committed:
autoconfig

CoolMineman committed:
add autoconfig translations

CoolMineman committed:
update to new fabric keybinding api and rebrand

CoolMineman committed:
make toggling use the action bar+change lang files

CoolMineman committed:
clean up imports

CoolMineman committed:
astromine support yeef

CoolMineman committed:
code cleanup

CoolMineman committed:
fix astromine version to not be a broken version

CoolMineman committed:
Fixes Fall Damage Negating Speed

CoolMineman committed:
fix the fix oops

CoolMineman committed:
fix astromine version for real

CoolMineman committed:
Speedometer, Enhanced Astromine Support, misc

CoolMineman committed:
fix version

CoolMineman committed:
1.16.5

CoolMineman committed:
fix slow fall

CoolMineman committed:
Merge branch '1.16-fabric' into master

squeek502 commented:
Nice work! Will test it out soon.

CoolMineman commented:
need to fix some merge issues first still

CoolMineman committed:
Revert QuakeClientPlayer as water was broken; merge seemed to have removed the pushMatrix

CoolMineman commented:
Seems to be working again now, QuakeClientPlayer seemed to have completely broken post merge so I copy and pasted my old version.

squeek502 commented:
Looks like there's still some stuff to work out with water movement (the same stuff I haven't figured out how to solve):

- While sharking, the player should be about waist-high in the water (in this branch you float above the water): https://youtu.be/pP_kIUJfPJU?t=24
- If the water is moving, the player's velocity should be affected (in this branch moving water doesn't affect the player): https://youtu.be/pP_kIUJfPJU?t=34

CoolMineman committed:
fix sharking height

CoolMineman commented:
Should be better now

CoolMineman committed:
sharking moving water

Siuolplex committed:
Add GitHub Actions support (#1)

Adds GitHub Action that compiles every commit and releases artifacts if successful.

Timwun commented:
Hey, whats up with this PR? 

@CoolMineman @squeek502

squeek502 commented:
Sharking is still not working the same as it was in previous versions. What I should probably do is just scratch sharking for the initial port and just release something; the sharking stuff probably needs a complete rewrite at this point.

Timwun commented:
This would be awesome! I think sharking is something important but I would just sacrifice it for an initial release.

Also I would love to see if the mod works in 1.17 as well. Fabric is already out. :D 

P.S.: Thanks for the mod! ❤️

davidalb97 commented:
To avoid server crashing at startup, add @Environment(EnvType.CLIENT) to ModQuakeMovement.drawSpeedometer() to the method signature.
